xmppserver/src/test/java/org/jivesoftware/openfire/group/DefaultGroupProviderTest.java

```diff
@@ -92,11 +91,16 @@ public void tearDown() throws Exception {
         // Reset static fields after use (to not confuse other test classes).
         // TODO: this ideally goes in a static @AfterClass method, but that's not supported in JUnit 3.
         for (String fieldName : Arrays.asList("INSTANCE", "provider")) {
-            Field field = GroupManager.class.getDeclaredField(fieldName);
+            final Field field = GroupManager.class.getDeclaredField(fieldName);
```

Comment:
Yuck. Is there way to avoid this? Perhaps a package-protected method to set the listeners, to be used for test purposes?

Reply:
That's trading in one yuck for another yuck. The test code is a bit of a mess already (with regards to not using `@AfterClass` and the concessions that it needs to do for that). I'm opting to keep the yuckiness in one class.